src/i18n/TextCollection/i18nTextCollector.php

```diff
@@ -883,7 +883,7 @@ public function collectFromCode($content, $fileName, Module $module)
                     $inTransFn = false;
                     $inConcat = false;
                     // Ensure key is valid before saving
-                    if (!empty($currentEntity[0])) {
+                    if (!empty($currentEntity[0]) && !str_ends_with($currentEntity[0], '.')) {
```

Comment:
I assume this is for 
> In assets File.php, an invalid key ending with "." with a value of "generic" was being collected

1. Shouldn't we have a PR in silverstripe/assets to fix that key?
2. Should we throw a warning when we find these instead of just ignoring them?
3. We should update [our PHPStan rule](https://github.com/silverstripe/silverstripe-standards/blob/1/src/PHPStan/TranslationFunctionRule.php) to capture this scenario so it doesn't happen again

Reply:
Discussed offline. We'll catch in PHPStan with https://github.com/silverstripe/silverstripe-standards/issues/16 later on